main/main.cpp

```diff
@@ -114,7 +114,12 @@
 
 #ifdef MODULE_GDSCRIPT_ENABLED
 #include "modules/gdscript/gdscript.h"
-#endif
+#ifdef TOOLS_ENABLED
+#ifndef GDSCRIPT_NO_LSP
```

Comment:
Nit: This could be unified as
```suggestion
#if defined(TOOLS_ENABLED) && !defined(GDSCRIPT_NO_LSP)
```
so there's a bit less nested `#ifdef`/`#endif`

Same below.